src/app/modules/consultations/consultation-list/consultation-list.component.ts

```diff
@@ -34,12 +36,33 @@ export class ConsultationListComponent implements OnInit {
     }
   }
 
-  constructor(private apollo: Apollo, private loader: LinearLoaderService, private errorService: ErrorService) { }
+  constructor(
+    private apollo: Apollo, 
+    private loader: LinearLoaderService, 
+    private errorService: ErrorService,
+    private userService: UserService
+    ) { }
 
   ngOnInit() {
+    this.checkUserSignedIn();
     this.fetchActiveConsultationList();
   }
+
+  checkUserSignedIn(){
+    this.userService.userLoaded$
+    .subscribe((exists: boolean) => {
+      if (exists) {
+        this.currentUser = this.userService.currentUser;
+      }
+    },
+    err => {
+      this.errorService.showErrorModal(err);
+    });
+  }
   
+  storeId(id: number){
```

Comment:
@Harshwardhan18 Keep this. The usage of this would be a bit different though. In the Auth endpoint, there will be a  field where you will need to send the API a callbackUrl so that they will know where to redirect the user to after authentication/email verification. Construct a callbackUrl to be sent to the API. callbackUrl will be- `https://www.civis.vote/consultations/:id/read`

Reply:
bro here we have callBackUrl for authSignUp mutation only not for login. So instead of that can I use the cookieServices to store the callBackUrl as used by nishanth in many places.

Reply:
@Harshwardhan18 Yep makes sense. Please also send the callBackUrl for authSignup mutation. For login, you can use this to redirect the user to the private consultation page

Reply:
@Harshwardhan18 You can remove this as well. Its being done in the Consultation profile page with the callback Url being stored in Local Storage

Reply:
@Harshwardhan18 Remove this

Reply:
@avikt already removed bro